### PR TITLE
[RFR][1LP] Add ensure_checked() and ensure_unchecked() functions to JSBaseEntity.

### PR DESCRIPTION
**Pull request by prichard77**
## Purpose or Intent.

- __Enhancement__ for selecting entities in the JSBaseEntity class. Using new functions ensure_checked() or ensure_unchecked() will make sure entities are unchecked before selecting them, and checked before un-selecting them.

### PRT Run
{{ pytest: --use-provider complete --long-running cfme/tests/infrastructure/test_host.py::test_infrastructure_hosts_crud
cfme/tests/infrastructure/test_host.py::test_infrastructure_hosts_refresh_multi
cfme/tests/infrastructure/test_host.py::test_compare_hosts_from_provider_allhosts }}

### CODE REVIEW

**Comment by prichard77** [comment]
The failures in PRT are not the tests that are affected by this update. I am still not sure what is wrong with my pytest command that is causing all tests in the modiule to get run.

File: widgetastic_manageiq/__init__.py [hunk 1]
```diff
@@ -3290,6 +3290,18 @@ def name(self):
         else:
             return getattr(self, "_name", None)
 
+    def ensure_checked(self):
+        if not self.is_checked:
```
Comment (valaparthvi) [inline widgetastic_manageiq/__init__.py]: Can you use `wait_for` here?

File: widgetastic_manageiq/__init__.py [hunk 2]
```diff
@@ -3290,6 +3290,18 @@ def name(self):
...
+            self.check()
+        if not self.is_checked:
+            raise WidgetOperationFailed(f"Checking of Entity {self} unsuccessful.")
+
+    def ensure_unchecked(self):
+        if self.is_checked:
```
Comment (valaparthvi) [inline widgetastic_manageiq/__init__.py]: Same comment as above.

File: widgetastic_manageiq/__init__.py (hunk 1 above)
Comment (prichard77) [reply widgetastic_manageiq/__init__.py]: I don't believe wait_for is needed here. I have tested this fairly well and the check functionality is essentially instantaneous.

**Pull request merged**